compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -640,26 +673,13 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
       val methodsBuilder = List.newBuilder[js.MethodDef]
       val jsNativeMembersBuilder = List.newBuilder[js.JSNativeMemberDef]
 
-      def gen(tree: Tree): Unit = {
-        tree match {
-          case EmptyTree => ()
-          case Template(_, _, body) => body foreach gen
```

Comment:
Not strictly dead code, but it would only happen for the top-level "recursive" call (we cannot have a `Template` inside a `Template`, and would *always* happen for the top-level call (the input is statically typed as `Template`, so we know that). The new code directly iterates over `impl.body` instead.